### DISCUSSION TITLE
Optimiziing RAM consumption

### DISCUSSION THREAD

**Discussion by volkerjaenisch** [eclipse-rdf4j/rdf4j#3353] [Q&A]
Dear RDF4J People!

We use a RDF4J database with around 10000 triples which is frequently updated with new data, but stays at an overall number of 10000 triples.
The java process uses 2.4 GB of RAM in the steady state. 
This is way to much RAM to accommodate the data and whatever sophisticated structure overhead.

We do not know much of Java memory configuration. There are the -Xms and -Xmx parameters available to reduce the RAM consumption.

But where do one have to set these Parameters (in Debian10) for RDF4J?

Any help appreciated

Volker

**Comment by barthanssens** [comment]
Hi Volker,

have you deployed the war on a Tomcat server as a systemd service ?
In that case you can alter / set the CATALINA_OPTS in /etc/systemd/system/tomcat.service 

See also https://www.digitalocean.com/community/tutorials/how-to-install-apache-tomcat-9-on-debian-10

Bart

**Reply by abrokenjester** [reply]
It's probably also worth noting that when you configure a JVM with 2G heap size (e.g. `-Xmx2G`), it will allocate those 2G, regardless of whether the app you run in the JVM requires all of it.
